### ISSUE TITLE
Nested property assignment evaluates incorrectly

### ISSUE THREAD

**Issue by svaarala** [svaarala/duktape#427]
(Originally issue filed in: https://github.com/svaarala/duktape-wiki/issues/59; see also http://stackoverflow.com/q/33489083/449161)

```
var NodeType;
(function (NodeType) {
    NodeType[NodeType["none"] = 0] = "none";
    NodeType[NodeType["text"] = 1] = "text";
})(NodeType || (NodeType = {}));
```

Duktape will resolve NodeType["text"] => 1 but not NodeType[1] => "text". The same code works fine in a browser such as Chrome.

**Comment by svaarala** [comment]
This bug seems to exist in Duktape 1.3.0 but seems to work correctly in master.

**Comment by svaarala** [comment]
Duktape 1.3.0 bytecode for the inner function:

```
; c0 "none"
; c1 {"type":"number","data":"0000000000000000","_value":0}
; c2 "text"
; c3 {"type":"number","data":"3ff0000000000000","_value":1}
00000 00000040   LDREG      r1, r0
00001 80c00056   PUTPROP    r1, c0, c1
00002 00000080   LDREG      r2, r0
00003 000040c0   LDREG      r3, r1
00004 8000c096   PUTPROP    r2, r3, c0
00005 00000040   LDREG      r1, r0
00006 81c08056   PUTPROP    r1, c2, c3
00007 00000080   LDREG      r2, r0
00008 0000c0c0   LDREG      r3, r3
00009 8100c096   PUTPROP    r2, r3, c2
00010 0000006f   RETURN     0x1, r0          ; have_retval
```

Bytecode in master:

```
; c0 "none"
; c1 {"type":"number","data":"0000000000000000","_value":0}
; c2 "text"
; c3 {"type":"number","data":"3ff0000000000000","_value":1}
00000 00000040   LDREG      r1, r0
00001 80c00056   PUTPROP    r1, c0, c1
00002 00000080   LDREG      r2, r0
00003 80404096   PUTPROP    r2, c1, c0
00004 00000040   LDREG      r1, r0
00005 81c08056   PUTPROP    r1, c2, c3
00006 00000080   LDREG      r2, r0
00007 8140c096   PUTPROP    r2, c3, c2
00008 0000002f   RETURN     0x0, r0
```

**Comment by svaarala** [comment]
Just to ensure this is not working in master because the assignment value is mapped to a constant, also checked that there's no issue with dynamic results:

```js
// Slightly modified
var obj2;
(function (obj2) {
    var t = '';
    var i = 0;
    obj2[obj2[t + "none"] = i + 0] = "none";
    obj2[obj2[t + "text"] = i + 1] = "text";
})(obj2 || (obj2 = {}));

[ 'none', 'text', 0, 1 ].forEach(function (k) {
   print(k, k in obj2, obj2[k]);
});
```

This works and the bytecode in master for that is:

```
; c0 ""
; c1 "none"
; c2 {"type":"number","data":"0000000000000000","_value":0}
; c3 "text"
; c4 {"type":"number","data":"3ff0000000000000","_value":1}
00000 00000042   LDCONST    r1, c0
00001 80000083   LDINT      r2, 0
00002 808040da   ADD        r3, r1, c1
00003 00000100   LDREG      r4, r0
00004 8100815a   ADD        r5, r2, c2
00005 0280c116   PUTPROP    r4, r3, r5
00006 00000180   LDREG      r6, r0
00007 80814196   PUTPROP    r6, r5, c1
00008 818040da   ADD        r3, r1, c3
00009 00000100   LDREG      r4, r0
00010 8200815a   ADD        r5, r2, c4
00011 0280c116   PUTPROP    r4, r3, r5
00012 00000180   LDREG      r6, r0
00013 81814196   PUTPROP    r6, r5, c3
00014 0000002f   RETURN     0x0, r0
```

**Comment by svaarala** [comment]
This was probably fixed in #381. Pull request #428 provides a testcase for this bug, and I added this to Duktape 1.3.1 fix list.

**Comment by fatcerberus** [comment]
That must be awesome, finding a bug and then realizing you already fixed it by accident a while ago. :)
